chapter09/ard/ex2.pl

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use v5.12;
+
+use File::Slurp;
+
+my @lines = read_file( 'patterns' );
+my @patterns;
+
+foreach my $line ( @lines ) {
+  chomp $line;
+  my $pattern;
+  eval {
+    $pattern = qr/$line/
+  };
```

Comment:
As the value of an `eval` block is the result of the last expression, this can be more concisely expressed as `my $pattern = eval { qr/$line/ };`, though I know some aren't into that.